packages/quantic/force-app/main/default/lwc/quanticResultPrintableUri/__tests__/quanticResultPrintableUri.test.js

```diff
@@ -0,0 +1,320 @@
+      resolve();
+    })
+  );
+
+  jest.spyOn(mockHeadlessLoader, 'getBueno').mockReturnValue(
+    new Promise(() => {
+      // @ts-ignore
+      global.Bueno = {
+        isString: jest
+          .fn()
+          .mockImplementation(
+            (value) =>
+              Object.prototype.toString.call(value) === '[object String]'
+          ),
+      };
+    })
+  );
+}
+
+const selectors = {
+  link: 'c-quantic-result-link',
+  pathItem: '[data-test="path__item"]',
+  pathExpandButton: '[data-test="path__expand-button"]',
+  errorMessage: '[data-test="error"]',
+};
+const exampleEngineId = 'example engine id';
+
+function createTestComponent(options) {
+  const element = createElement('c-quantic-result-printable-uri', {
+    is: QuanticResultPrintableUri,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function generateExampleUri(id) {
+  return `https://uri${id}.com/`;
+}
+
+function generateResultParents(numberOfParents) {
+  let parents = '';
+  for (let i = 1; i <= numberOfParents; i++) {
+    parents += `<parent name="parent${i}" uri="${generateExampleUri(i)}" />`;
+  }
+  return `<parents>${parents}</parents>`;
+}
+
+function shouldProperlyDisplayAllPathItems(pathItems, numberOfItems) {
+  expect(pathItems).not.toBeNull();
+  expect(pathItems.length).toBe(numberOfItems);
+  pathItems.forEach((item, index) => {
+    expect(item.textContent).toBe(`parent${index + 1}`);
+    expect(item.href).toBe(`${generateExampleUri(index + 1)}`);
+  });
+}
+
+describe('c-quantic-result-printable-uri', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  }
+
+  beforeEach(() => {
+    console.error = jest.fn();
+    mockHeadlessAndBueno();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when displaying the result path', () => {
+    describe('when the number of parents is lower than the maxNumberOfParts property', () => {
+      const numberOfParents = 2;
+      const exampleResultWithParents = {
+        raw: {
+          parents: generateResultParents(numberOfParents),
+        },
+      };
+      it('should properly display all the result path', async () => {
+        const element = createTestComponent({
+          result: exampleResultWithParents,
+          engineId: exampleEngineId,
+          maxNumberOfParts: 3,
+        });
+        await flushPromises();
+
+        const pathItems = element.shadowRoot.querySelectorAll(
+          selectors.pathItem
+        );
+        shouldProperlyDisplayAllPathItems(pathItems, numberOfParents);
```

Comment:
I thought we no longer wanted test without assertions and that the expect should be inside the test for readability no? @erocheleau @mmitiche

Reply:
I personally prefer having the expect in the tests, even if this means having duplicate the code.

But as I said in the answer to @erocheleau 's comment, it's more like a preference.

Reply:
I mean we do need expects in each test :D 

But if you have a series of like 4 expects that repeat everywhere, you can put them in a function to avoid duplicating those lines of code many times.

Reply:
That's why in my suggestion I left the `expect(pathItems).not.toBeNull();` outside the function :) so we wouldn't have this pmd warning hehe